plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -101,6 +101,9 @@ fu_plugin_dell_dock_probe(FuPlugin *plugin, FuDevice *proxy, GError **error)
 	/* create TBT endpoint if Thunderbolt SKU and Thunderbolt link inactive */
 	if (fu_dell_dock_ec_needs_tbt(FU_DEVICE(ec_device))) {
 		g_autoptr(FuDellDockTbt) tbt_device = fu_dell_dock_tbt_new(proxy);
+		fu_device_set_context(FU_DEVICE(tbt_device), ctx);
+		fu_device_add_guid(FU_DEVICE(tbt_device),
+				   fwupd_guid_hash_string(DELL_DOCK_TBT_INSTANCE_ID));
```

Comment:
this leaks the string; I really think you should use `fu_device_add_instance_id(FU_DEVICE(tbt_device), DELL_DOCK_TBT_INSTANCE_ID)` instead -- although it might mean calling `fu_device_setup()` manually.

Reply:
We need the `parent` device prepared prior to running `setup`, the flow would look like `add_guid` > `add_child` > `create_node` for `open` and `setup`. I can make a change to `autofree` the string which is introduced by hash.